A stab at using vectors again.

hassila opened the pull request:
Please have a look at this PR, numbers looks good but needs a second eye for verifying correctness.
```
decode_direct4_class
=================================
2527 ms encode
0 ms decode
162 ms use
162 ms decode+use
=================================

decode_direct4_struct
=================================
2545 ms encode
0 ms decode
152 ms use
152 ms decode+use
=================================

decode_direct4_struct_with_vector
=================================
2525 ms encode
0 ms decode
136 ms use
136 ms decode+use
=================================
```

hassila committed:
A stab at using vectors again.

mzaks commented on FBTest/FBReaderCommon.swift:
```diff
@@ -186,82 +186,96 @@ public extension FBReader {
+            return T(reader: reader, myOffset: reader.getVectorOffsetElement(vectorOffset: myOffset, index: i))!
+    }
+    /*
+    public subscript(i : Int) -> T? {
```
I would rather go with this subscript.
I generally made FlatBuffersSwift much more fault tolerant specifically on read.
Rationale is if some one sends you corrupt buffer intentionally, your application should not crash.
Specifically important for BackEnd applications

hassila replied:
The one returning an optional? Or the original one?

mzaks replied:
Returning optional is better the user can force unwrap it if they sure the buffer is not corrupted

hassila replied:
That's fine, it was my first implementation - it performs the same essentially.

mzaks replied:
👍

hassila committed:
Switch to subscript returning optional according to review feedback, new results are in line:

decode_direct4_struct
=================================
2559 ms encode
0 ms decode
156 ms use
156 ms decode+use
=================================

decode_direct4_struct_with_vector
=================================
2526 ms encode
0 ms decode
136 ms use
136 ms decode+use
=================================

hassila commented:
Ok, f7a733c adopts that pattern. Interestingly enough the latest revision using vectors actually gives a consistent performance improvement of ~10%.
